services/auth-service/src/main/java/com/sourcefuse/jarc/services/authservice/oauth2/providers/OAuth2SignupProvider.java

```diff
@@ -0,0 +1,116 @@
+package com.sourcefuse.jarc.services.authservice.oauth2.providers;
+
+import com.sourcefuse.jarc.core.enums.RoleKey;
+import com.sourcefuse.jarc.services.authservice.dtos.RegisterDto;
+import com.sourcefuse.jarc.services.authservice.enums.AuthErrorKeys;
+import com.sourcefuse.jarc.services.authservice.enums.AuthProvider;
+import com.sourcefuse.jarc.services.authservice.models.Role;
+import com.sourcefuse.jarc.services.authservice.models.Tenant;
+import com.sourcefuse.jarc.services.authservice.models.User;
+import com.sourcefuse.jarc.services.authservice.oauth2.user.OAuth2UserInfo;
+import com.sourcefuse.jarc.services.authservice.repositories.RoleRepository;
+import com.sourcefuse.jarc.services.authservice.repositories.TenantRepository;
+import com.sourcefuse.jarc.services.authservice.services.UserService;
+import com.sourcefuse.jarc.services.authservice.specifications.RoleSpecification;
+import com.sourcefuse.jarc.services.authservice.specifications.TenantSpecification;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpServerErrorException;
+
+@Service
+@RequiredArgsConstructor
+public class OAuth2SignupProvider {
+
+  private final TenantRepository tenantRepository;
+  private final UserService userService;
+  private final RoleRepository roleRepository;
+
+  public User provide(OAuth2UserInfo oAuth2UserInfo, AuthProvider provider) {
+    Optional<Tenant> tenant =
+      this.tenantRepository.findOne(TenantSpecification.byKey("master"));
```

Comment:
can we move this constant "master" to some Constant file as we are using it in multiple places.